### PR TITLE
140 mongo

### PR DESCRIPTION
**Pull request by twardnw**

### CODE REVIEW

**Comment by Schnitzel** [comment]
Looks good, some changes:
- for the env variables we use a new feature of kubernetes 1.6 to import all keys from a configmap as env variables:
https://github.com/amazeeio/lagoon/blob/5da1856ba13eabd66df31d45a59c35cfdc4cafd3/images/oc-build-deploy-dind/openshift-templates/redis/template.yml#L73-L75
- is there a reason that we use the centos instead of alpine with mongo installed?

**Review by Schnitzel — CHANGES_REQUESTED** [review]
see comment :)

**Comment by twardnw** [comment]
Well, same story as Varnish, there was a mongodb package hiding in APK Edge, have that added now, also updated to use the env var map too.
 
@Schnitzel please review again? :)

File: images/mongo/docker-entrypoint [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec "$@"
```
Comment (Schnitzel) [inline images/mongo/docker-entrypoint]: this is already defined in the main entrypoints.sh file:
https://github.com/amazeeio/lagoon/blob/master/images/commons/lagoon/entrypoints.sh#L15

so we don't need that file at all :)

File: images/mongo/yum.mongodb-org-3.4.repo [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[mongodb-org-3.4]
+name=MongoDB Repository
+baseurl=https://repo.mongodb.org/yum/redhat/$releasever/mongodb-org/3.4/x86_64/
+gpgcheck=1
+enabled=1
+gpgkey=https://www.mongodb.org/static/pgp/server-3.4.asc
```
Comment (Schnitzel) [inline images/mongo/yum.mongodb-org-3.4.repo]: yum in alpine? :) 
guess we can remove that file.

File: images/oc-build-deploy-dind/openshift-templates/mongo/template.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    description: git hash sha of the current deployment
+    required: true
+  - name: ROUTER_URL
+    description: URL of the Router
+    required: true
+objects:
+- apiVersion: v1
+  kind: PersistentVolumeClaim
+  metadata:
+    name: ${SERVICE_NAME}-persistent
```
Comment (Schnitzel) [inline images/oc-build-deploy-dind/openshift-templates/mongo/template.yml]: we don't call the PVCs `-persistent ` anymore, as their name already suggests that they are persistent, just the `${SERVICE_NAME}`

File: images/oc-build-deploy-dind/openshift-templates/mongo/template.yml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+              tcpSocket:
+                port: 27017
+              initialDelaySeconds: 15
+              timeoutSeconds: 1
+          envFrom:
+          - configMapRef:
+             name: lagoon-env
+          volumeMounts:
+            - name: ${SERVICE_NAME}-persistent
+              mountPath: ${PERSISTENT_STORAGE_PATH}
```
Comment (Schnitzel) [inline images/oc-build-deploy-dind/openshift-templates/mongo/template.yml]: mongo has a predefined storage path, so it's not likely that a user wants to inject another path, let's define it hardcoded, see https://github.com/amazeeio/lagoon/blob/master/images/oc-build-deploy-dind/openshift-templates/mariadb/template.yml#L89 as an example

File: images/oc-build-deploy-dind/openshift-templates/mongo/template.yml [hunk 5]
```diff
@@ -0,0 +1,117 @@
+apiVersion: v1
+kind: Template
+metadata:
+  creationTimestamp: null
+  name: amazeeio-openshift-template-mongo
+parameters:
```
Comment (Schnitzel) [inline images/oc-build-deploy-dind/openshift-templates/mongo/template.yml]: these seem not up to date, see https://github.com/amazeeio/lagoon/blob/master/images/oc-build-deploy-dind/openshift-templates/mariadb/template.yml#L6 as an example

**Review by Schnitzel — CHANGES_REQUESTED** [review]

**Review by Schnitzel — APPROVED** [review]

**Pull request merged**